chapter_linear-classification/softmax-regression-scratch.md

```diff
@@ -244,7 +245,20 @@ y = tf.constant([0, 2])
 tf.boolean_mask(y_hat, tf.one_hot(y, depth=y_hat.shape[-1]))
 ```
 
+:begin_tab:`pytorch, mxnet, tensorflow`
 Now we can (**implement the cross-entropy loss function**) by averaging over the logarithms of the selected probabilities.
+:end_tab:
+
+:begin_tab:`jax`
+Now we can (**implement the cross-entropy loss function**) by averaging over the logarithms of the selected probabilities.
+
+Note that to make use of `jax.jit` to speed up JAX implementation, and
+to make sure `loss` is a pure function, the `cross_entropy` function is re-defined
+inside the `loss` to avoid usage of any global variables or functions
+which may render the `loss` function impure.
+
```

Comment:
```suggestion
```

Reply:
Since readers may have doubts on "pure", merge these two paragraphs to make the following explanations closer to where "pure"/"impure" are used.